### PR TITLE
feat(if): add static Observable.if creation operator.

### PR DESCRIPTION
**Pull request by trxcllnt**
Addresses #1343

### CODE REVIEW

File: src/Rx.KitchenSink.ts [hunk 1]
```diff
@@ -39,8 +39,10 @@ import './add/observable/fromArray';
 import './add/observable/fromEvent';
 import './add/observable/fromEventPattern';
 import './add/observable/fromPromise';
+import './add/observable/if';
```
Comment (kwonoj) [inline src/Rx.KitchenSink.ts]: is this for `kitchensink` only? not in `rx`?

Comment (trxcllnt) [inline src/Rx.KitchenSink.ts]: @kwonoj in Rx4 `Observable.if` is in the "experimental" package, so I only put it into KitchenSink, though I'm still left wondering why we distinguish between "core" and "kitchen sink" operators in the first place.

File: src/observable/IfObservable.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    if (errorHappened) {
+      subscriber.error(error);
+    } else if (result && thenSource) {
+      return thenSource.subscribe(subscriber);
+    } else if (elseSource) {
+      return elseSource.subscribe(subscriber);
+    } else {
+      subscriber.complete();
+    }
```
Comment (luisgabriel) [inline src/observable/IfObservable.ts]: What about moving this if to a separate function? I know this is not a very performance sensitive operator, but we can have improvements anyway.

Comment (trxcllnt) [inline src/observable/IfObservable.ts]: Sure, that's alright with me.

Comment (kwonoj) [inline src/observable/IfObservable.ts]: @trxcllnt , would you be update PR as suggested, or proceed to check in as-is?

**Comment by blesh** [comment]
@trxcllnt I think this needs rebased/refactored to move the tests to TypeScript.

**Comment by trxcllnt** [comment]
@blesh done

**Comment by blesh** [comment]
@trxcllnt this still appears to be in need of a rebase. I'm sure that something just moved around you, but can you hit it again? After that, myself or @kwonoj will merge it in ASAP.

**Comment by kwonoj** [comment]
I'll check this in today. :)

**Comment by kwonoj** [comment]
Merged with f7ff7ec, thanks @trxcllnt including several rebase :)

**Pull request merged**